edxval/api.py

```diff
@@ -296,11 +342,47 @@ def get_video_info_for_course_and_profile(course_id, profile_name):
         raise ValInternalError(error_message)
 
     # DRF serializers were causing extra queries for some reason...
-    return {
-        enc_vid.video.edx_video_id: {
-            "url": enc_vid.url,
-            "file_size": enc_vid.file_size,
-            "duration": enc_vid.video.duration,
-        }
-        for enc_vid in encoded_videos
-    }
+    return_dict = {}
+    for enc_vid in encoded_videos:
+        return_dict.setdefault(enc_vid.video.edx_video_id, {}).update(
+            {
+                enc_vid.profile.profile_name: {
+                    "url": enc_vid.url,
+                    "file_size": enc_vid.file_size,
+                    "duration": enc_vid.video.duration,
+                }
+            }
+        )
+
+    return return_dict
+
+
+def rerun_course(old_course_id, new_course_id):
+    """
+    Adds the new_course_id to the videos taken from the old_course_id
+
+    Args:
+        old_course_id: The original course_id
+        new_course_id: The new course_id of the rerun
+    """
+    if old_course_id == new_course_id:
+        raise ValCannotCreateError(
+            "Both course_id's are the same: {}".format(new_course_id)
+        )
+    if Video.objects.filter(courses__course_id=unicode(new_course_id)):
+        raise ValCannotCreateError(
+            "New course id already exists: {}".format(new_course_id)
+        )
+    videos = Video.objects.filter(courses__course_id=unicode(old_course_id))
+
+    if len(videos) == 0:
+        raise ValVideoNotFoundError(
+            "Cannot find any videos for given course_id: {}".format(old_course_id)
+        )
+
+    for video in videos:
+        CourseVideo.objects.create(
```

Comment:
If the restriction against calling this function for a pre-existing course is removed, this should change to `get_or_create`